cpp/src/Ice/ConnectionFactory.cpp

```diff
@@ -1239,11 +1235,11 @@ IceInternal::IncomingConnectionFactory::flushAsyncBatchRequests(
 {
     list<ConnectionIPtr> c = connections(); // connections() is synchronized, so no need to synchronize here.
 
-    for (list<ConnectionIPtr>::const_iterator p = c.begin(); p != c.end(); ++p)
+    for (auto& p : c)
```

Comment:
Seem to be lots of these. 
```suggestion
    for (const auto& p : c)
```